Anature/src/application/anatures/AnatureBase.java

```diff
@@ -0,0 +1,729 @@
+		}
+
+		int adjustment = (int) (defenseAdjustment * mDefense);
+
+		mTempDefense = mTempDefense + adjustment;
+	}
+
+	public void adjustSpecialAttack(double specialAttackAdjustment)
+	{
+		if(specialAttackAdjustment < -1 || specialAttackAdjustment > 1)
+		{
+			throw new IllegalArgumentException("Passed value \"tempSpecialAttackAdjustment\" was below -1 or above 1.");
+		}
+
+		int adjustment = (int) (specialAttackAdjustment * mSpecialAttack);
+
+		mTempSpecialAttack = mTempSpecialAttack + adjustment;
+	}
+
+	public void adjustSpecialDefense(double specialDefenseAdjustment)
+	{
+		if(specialDefenseAdjustment < -1 || specialDefenseAdjustment > 1)
+		{
+			throw new IllegalArgumentException("Passed value \"tempSpecialDefenseAdjustment\" was below -1 or above 1.");
+		}
+
+		int adjustment = (int) (specialDefenseAdjustment * mSpecialDefense);
+
+		mTempSpecialDefense = mTempSpecialDefense + adjustment;
+	}
+
+	public void adjustSpeed(double speedAdjustment)
+	{
+		if(speedAdjustment < -1 || speedAdjustment > 1)
+		{
+			throw new IllegalArgumentException("Passed value \"tempSpeedAdjustment\" was below -1 or above 1.");
+		}
+
+		int adjustment = (int) (speedAdjustment * mSpeed);
+
+		mTempSpeed = mTempSpeed + adjustment;
+	}
+
+	public void adjustAccuracy(double accuracyAdjustment)
+	{
+		if(accuracyAdjustment < -1 || accuracyAdjustment > 1)
+		{
+			throw new IllegalArgumentException("Passed value \"accuracyAdjustment\" was below -1 or above 1.");
+		}
+
+		int adjustment = (int) (accuracyAdjustment * mAccuracy);
+
+		mTempAccuracy = mTempAccuracy + adjustment;
+	}
+
+	public void updateName(String name)
+	{
+		setName(name);
+	}
+
+	public void updateStatus(StatusEffects status)
+	{
+		setStatus(status);
+	}
+
+	public void updateCurrentHitPoints(int hitPoints)
+	{
+		setCurrentHitPoints(hitPoints);
+	}
+
+	public ArrayList<Type> getTypes()
+	{
+		ArrayList<Type> types = new ArrayList<Type>();
+
+		if(!getPrimaryType().equals(Type.NotSet))
+		{
+			types.add(getPrimaryType());
+		}
+		if(!getSecondaryType().equals(Type.NotSet))
+		{
+			types.add(getSecondaryType());
+		}
+
+		return types;
+	}
+
+	public void resetTempStats()
+	{
+		mTempAttack = 0;
+		mTempSpecialAttack = 0;
+		mTempDefense = 0;
+		mTempSpecialDefense = 0;
+		mTempSpeed = 0;
+		mTempAccuracy = 0;
+	}
+
+	public void takeDamage(int damage)
+	{
+		mCurrentHitPoints -= damage;
+
+		if(mCurrentHitPoints < 0)
+			mCurrentHitPoints = 0;
+	}
+
+	public String healAnature(int healAmount)
+	{
+		int hitPointsAfterHeal = getCurrentHitPoints() + healAmount;
+		if(healAmount == Integer.MAX_VALUE || hitPointsAfterHeal > getTotalHitPoints())
+		{
+			setCurrentHitPoints(getTotalHitPoints());
+			return " was healed completely!";
+		}
+
+		setCurrentHitPoints(hitPointsAfterHeal);
+		return " was healed " + healAmount + " hp.";
+	}
+
+	public double getHitPointsPercent()
+	{
+		return mCurrentHitPoints / mTotalHitPoints;
+	}
```

Comment:
Due to the fact that these are ints, this will only every resolve as 0 or 1 (because it will do the conversion to a double after the result). With how it is, the trainer will use a potion as soon as they take any damage.
Here is what I did to fix it:
return (double) mCurrentHitPoints / mTotalHitPoints